src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -88,11 +91,6 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
         // To avoid allocation, return a null scope if the logger is not on at least to some degree.
         if (loggingEnabled)
         {
-            // Scope may be relevant for a different level of logging, so we always create it
-            // see: https://github.com/aspnet/Hosting/pull/944
-            // Scope can be null if logging is not on.
-            context.Scope = Log.RequestScope(_logger, httpContext);
```

Comment:
I think my main concern about switching from logger scope to Activity is checking whether typical tooling that users have available today for log viewing can do a good job correlating Activity tags with log messages. There is no reason that tools couldn't do it, but it may be something that wasn't important previously so the tools didn't invest the energy into it. On the one hand if do this then it will provide motivations for tools to support it better, but it might leave users in a worse place until the tools catch up. A more conservative route would be adding the new info to Activity and tell tooling folks that we plan to remove the data from logger scope in the future to encourage them to prepare.

Reply:
I think 2 things can help mitigate the breaking change:
1. We support pushing activity tags to the logging activity scope (the same way we do for the built in ones)
2. Assuming we did this, it would flow to logger provider implementations that supported the IExternalScopeProvider

But it would mean that we aren't explicitly adding a logging scope so if the ILoggerProvider doesn't support IExternalScopeProvider, it would be more visibly breaking.

Reply:
I don't have any background knowledge on how often users correlate log messages using those tags we are moving, so I'm mostly relying on you to decide how breaking this is. I could see how an option that includes all Activity tags in ExternalScopeProvider could be useful, but if you think making this change safe requires support for that option first then we probably want to put this PR on hold until then.

Reply:
Agree, I'll leave it as a draft. I think we should open up a new issue for "Activity tags as log scope entries".

Reply:
We already support logging Activity tags in https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.activitytrackingoptions?view=dotnet-plat-ext-6.0

Reply:
Ah, nice!

Reply:
I'm pretty convinced this scope useful by default and can be easily added back via middleware. The cost we're paying for the benefit people get by default is too high (many don't even know how to look at these logging scopes). I think now that the ability to associate logs with the activity exists and is on by default. I feel pretty confident breaking this.